### PR TITLE
Fixes to Request Context Change section

### LINKED ISSUE #144: May 2019 Ballot Comment: Add the traditional table describing the message fields for Request Context Change.
## May 2019 Ballot Comment: Add the traditional table describing the message fields for Request Context Change.

Submitted by Ricardo Quintano Neira 
**Chapter/section:**  Request Context Change
**Url:** https://fhircast.hl7.org/specification/May2019Ballot/index.html
**Type:** NEG :exclamation: Clarification

**Summary:** Add the traditional table describing the message fields for Request Context Change.

**Comment:** I suggest adding the the traditional table describing the message fields for Request Context Change. 

See table example in Event Notification Request Details section.

---
_This issue was imported by @hl7-fhircast-bot from the [consolidated FHIRcast May 2019 ballot spreadsheet](https://bit.ly/fhircast-May-2019-ballot-comments)._

### PR DESCRIPTION
**Pull request by wmaethner**
Fixes #144, #170, #209

### CODE REVIEW

File: docs/specification/STU1.md [hunk 1]
```diff
@@ -312,11 +312,24 @@ Authorization: Bearer i8hweunweunweofiwweoijewiwe
...
 
-Once a requested context change is accepted, the Hub MUST broadcast the context notification to all subscribers, including the original requestor. 
+Once a requested context change is accepted, the Hub SHALL broadcast the context notification to all subscribers, including the original requestor. 
 
 ### Request Context Change Request
+###### Request Context Change Parameters
+Field | Optionality | Type | Description
+--- | --- | --- | ---
+`timestamp` | Required | *string* | ISO 8601-2 timestamp in UTC describing the time at which the event occurred with subsecond accuracy. 
+`id` | Required | *string* | Event identifier used to recognize retried notifications. This id SHALL be generated by and unique for the subscriber, for example a GUID.
```
Comment (lbergnehr) [inline docs/specification/STU1.md]: > This id SHALL be generated by and unique for the subscriber, for example a GUID.

This sentence probably needs a little rewording. Perhaps "This id SHALL be uniquely generated by the subscriber and could be a GUID.".

File: docs/specification/STU1.md [hunk 2]
```diff
@@ -312,11 +312,24 @@ Authorization: Bearer i8hweunweunweofiwweoijewiwe
...
+`event` | Required | *object* | A json object describing the event. See [below](#request-context-change-event-object-parameters).
+
+###### Request Context Change Event Object Parameters
+Field | Optionality | Type | Description
+--- | --- | --- | ---
+`hub.topic` | Required | string | The topic session uri given in the subscription request.
```
Comment (lbergnehr) [inline docs/specification/STU1.md]:
```suggestion
`hub.topic` | Required | string | The topic session URI given in the subscription request. 
```

File: docs/specification/STU1.md [hunk 3]
```diff
@@ -312,11 +312,24 @@ Authorization: Bearer i8hweunweunweofiwweoijewiwe
...
+`timestamp` | Required | *string* | ISO 8601-2 timestamp in UTC describing the time at which the event occurred with subsecond accuracy. 
+`id` | Required | *string* | Event identifier used to recognize retried notifications. This id SHALL be generated by and unique for the subscriber, for example a GUID.
+`event` | Required | *object* | A json object describing the event. See [below](#request-context-change-event-object-parameters).
+
+###### Request Context Change Event Object Parameters
+Field | Optionality | Type | Description
+--- | --- | --- | ---
+`hub.topic` | Required | string | The topic session uri given in the subscription request. 
+`hub.event`| Required | string | The event that triggered this request for the subscriber, taken from the list of events from the subscription request.
+`context` | Required | array | An array of named FHIR objects corresponding to the user's context after the given event has occurred. Common FHIR resources are: Patient, Encounter, ImagingStudy and List. The subscriber SHALL only include FHIR resources that are authorized to be accessed with the existing OAuth2 access_token.
```
Comment (lbergnehr) [inline docs/specification/STU1.md]: Perhaps
> ... OAuth2 `access_token`.

or

> ... OAuth2 access token.

File: docs/specification/STU1.md (hunk 1 above)
Comment (wmaethner) [reply docs/specification/STU1.md]: Updated

File: docs/specification/STU1.md (hunk 2 above)
Comment (wmaethner) [reply docs/specification/STU1.md]: Updated

File: docs/specification/STU1.md (hunk 3 above)
Comment (wmaethner) [reply docs/specification/STU1.md]: Updated

File: docs/specification/STU1.md [hunk 4]
```diff
@@ -312,11 +312,24 @@ Authorization: Bearer i8hweunweunweofiwweoijewiwe
...
 
-Once a requested context change is accepted, the Hub MUST broadcast the context notification to all subscribers, including the original requestor. 
+Once a requested context change is accepted, the Hub SHALL broadcast the context notification to all subscribers, including the original requestor. 
 
 ### Request Context Change Request
+###### Request Context Change Parameters
+Field | Optionality | Type | Description
+--- | --- | --- | ---
+`timestamp` | Required | *string* | ISO 8601-2 timestamp in UTC describing the time at which the event occurred with subsecond accuracy. 
+`id` | Required | *string* | Event identifier used to recognize retried notifications. This id SHALL be uniquely generated by the subscriber and could be a GUID.
```
Comment (isaacvetter) [inline docs/specification/STU1.md]:
```suggestion
`id` | Required | *string* | Event identifier used to recognize retried notifications. This id SHALL be uniquely generated by the subscriber and could be a GUID. Following an accepted context change request, the hub MAY re-use this value in the broadcasted event notifications.
```

@wmaethner - I think that we need this additional sentence to fully address #170.

**Pull request merged**